Zend/zend_enum.c

```diff
@@ -41,9 +41,17 @@ zend_object *zend_enum_new(zval *result, zend_class_entry *ce, zend_string *case
 	zend_object *zobj = zend_objects_new(ce);
 	ZVAL_OBJ(result, zobj);
 
-	ZVAL_STR_COPY(OBJ_PROP_NUM(zobj, 0), case_name);
+	zval *zname = OBJ_PROP_NUM(zobj, 0);
+	ZVAL_STR_COPY(zname, case_name);
+	/* ZVAL_COPY does not set Z_PROP_FLAG, this needs to be cleared to avoid leaving IS_PROP_REINITABLE set */
```

Comment:
This still confused me because I didn't understand why this flag was set in the first place. Turns out that `zend_objects_new()` doesn't `calloc`, or `memset` the properties. Maybe this could be documented accordingly.

Reply:
This looks fine. `object_properties_init` isn't called for enums, but even if it was, `IS_PROP_UNINIT` would be set to `1` because we don't have a default value in the property declaration.